src/main/java/org/HdrHistogram/AbstractHistogram.java

```diff
@@ -296,15 +296,15 @@ private void init(final long lowestDiscernibleValue,
          */
         final long largestValueWithSingleUnitResolution = 2 * (long) Math.pow(10, numberOfSignificantValueDigits);
 
-        unitMagnitude = (int) Math.floor(Math.log(lowestDiscernibleValue)/Math.log(2));
+        unitMagnitude = (int) (Math.log(lowestDiscernibleValue)/Math.log(2));
         unitMagnitudeMask = (1 << unitMagnitude) - 1;
 
         // We need to maintain power-of-two subBucketCount (for clean direct indexing) that is large enough to
         // provide unit resolution to at least largestValueWithSingleUnitResolution. So figure out
         // largestValueWithSingleUnitResolution's nearest power-of-two (rounded up), and use that:
         int subBucketCountMagnitude = (int) Math.ceil(Math.log(largestValueWithSingleUnitResolution)/Math.log(2));
-        subBucketHalfCountMagnitude = ((subBucketCountMagnitude > 1) ? subBucketCountMagnitude : 1) - 1;
-        subBucketCount = (int) Math.pow(2, (subBucketHalfCountMagnitude + 1));
+        subBucketHalfCountMagnitude = subBucketCountMagnitude - 1;
+        subBucketCount = 1 << subBucketCountMagnitude;
```

Comment:
The rest of the codebase seems to consistently use shifts to express `2^k`, so this one stood out, and now is cast-free. (Maybe the VM optimizes away casts in `Math.pow`?)